crates/uv/src/commands/project/mod.rs

```diff
@@ -236,17 +236,35 @@ pub(crate) fn init_environment(
 }
 
 /// Update a [`PythonEnvironment`] to satisfy a set of [`RequirementsSource`]s.
+#[allow(clippy::too_many_arguments)]
 pub(crate) async fn update_environment(
     venv: PythonEnvironment,
     requirements: &[RequirementsSource],
-    index_locations: &IndexLocations,
+    settings: &InstallerSettings,
+    preview: PreviewMode,
     connectivity: Connectivity,
+    concurrency: Concurrency,
+    native_tls: bool,
     cache: &Cache,
     printer: Printer,
-    preview: PreviewMode,
 ) -> Result<PythonEnvironment> {
-    // TODO(zanieb): Support client configuration
-    let client_builder = BaseClientBuilder::default().connectivity(connectivity);
+    // Extract the project settings.
+    let InstallerSettings {
+        index_locations,
+        index_strategy,
+        keyring_provider,
+        resolution,
+        prerelease,
+        config_setting,
+        exclude_newer,
+        link_mode,
+        compile_bytecode,
+    } = settings;
+
+    let client_builder = BaseClientBuilder::new()
+        .connectivity(connectivity)
+        .native_tls(native_tls)
+        .keyring(*keyring_provider);
```

Comment:
Separate, but I wonder if we should have a `Settings::as_client_builder` or `BaseClientBuilder::from_settings`